src/SpecialCreateBranch.php

```diff
@@ -52,40 +58,47 @@ protected function getFormFields() {
 		return $formDescriptor;
 	}
 
-	public function onSubmit(array $data) {
-		$this->createdPageTitle = Title::newFromText($data[ 'BranchName' ]);
-
-		$text = "Create branch request parameters:\n";
-		foreach ($data as $key => $val) {
-			$text .= "* $key: $val\n";
-		}
-
-		return $this->createPage($this->createdPageTitle, $text);
-	}
+	public function onSubmit( array $data ) {
+		$branchName = $data['BranchName'];
 
-	private function createPage(Title $title, string $wikitext = '') {
+		$title = Title::newFromText( $branchName );
 		if ( $title->isKnown() ) {
 			return Status::newFatal( 'createbranch-error-already-exists' );
 		}
 		if ( !$title->canExist() ) {
 			return Status::newFatal( 'createbranch-error-invalid-name' );
 		}
+		$this->branchPageTitle = $title;
 
+		// create branch remotely
+		try {
+			$branchId = LakatStorageStub::getInstance()->createBranch( $branchName, $data );
+		} catch ( Exception $e ) {
+			return Status::newFatal( 'createbranch-error-remote' );
+		}
+
+		// create branch root page
 		$page = MediaWikiServices::getInstance()->getWikiPageFactory()->newFromTitle( $title );
-		$text = ContentHandler::makeContent( $wikitext, null, LakatContent::MODEL_ID);
+
+		$json = FormatJson::encode( [ 'BranchId' => $branchId ] + $data );
+		$content = ContentHandler::makeContent( $json, $title, CONTENT_MODEL_JSON );
+		if ( !$content->isValid() ) {
+			throw new LogicException( 'Json parsing failed' );
+		}
+
 		$comment = CommentStoreComment::newUnsavedComment(
 			wfMessage( 'createbranch-revision-comment' )->inContentLanguage()->text()
 		);
 
 		$page->newPageUpdater( $this->getUser() )
-			->setContent( SlotRecord::MAIN, $text )
+			->setContent( SlotRecord::MAIN, ContentHandler::makeContent('Branch root page', $title) )
+			->setContent( 'lakat', $content )
 			->saveRevision( $comment );
 
```

Comment:
How does this local file storage solution work? 
SlotRecord::MAIN is a global variable?

Reply:
To clarify, this code is related to metadata stored as part of wiki page content. Just to distinguish from local file storage that I created as quick temporary replacement of remote distributed db.

`SlotRecord::MAIN` is a `SlotRecord `class constant, it's just a string `'main'`. In mediawiki slot `main` is dedicated to the main wiki page content. Main slot is required for any wiki page.